http-clients/src/main/java/org/triplea/http/client/web/socket/GenericWebSocketClient.java

```diff
@@ -57,7 +57,7 @@ public void registerListenerAndConnect(final Consumer<ServerMessageEnvelope> mes
             throwable -> {
               log.log(
                   Level.SEVERE, "Unexpected exception completing websocket connection", throwable);
-              return false;
+              return null;
```

Comment:
This block is notable as the error handling is pretty screwed up to begin with. We have an error handler passed to the connect function, yet we still choose to have our own custom error handling on this out layer. Really the inner call `connect` should take care of the `exceptionally` case and invoke the error handler. As-is, it seems like the error-handler is a violation of DRY (expressing the same idea in multiple places) or arguably SRP.